presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Rule.java

```diff
@@ -46,4 +48,40 @@ default boolean isEnabled(Session session)
 
         Session getSession();
     }
+
+    class Result
+    {
+        @Deprecated
+        public static Result fromOptional(Optional<PlanNode> transformedPlan)
```

Comment:
Why not to overload `replace/add` method? I guess the same `fromOptional` method would be for trait, so maybe it's better to group them by semantics (e:g. `addPlanNode`, `addTrait`)?

Reply:
I will remove this method.